src/LiquidityManager.sol

```diff
@@ -0,0 +1,189 @@
+    function PT() external view returns (address);
+    function YT() external view returns (address);
+    function reserveX() external view returns (uint256);
+    function reserveY() external view returns (uint256);
+    function prepareAllocate(uint256 deltaX, uint256 deltaY, PYIndex index) external view returns (uint256 deltaXWad, uint256 deltaYWad, uint256 deltaLiquidity, uint256 lptMinted);
+    function swapExactPtForSy(uint256 amountIn, uint256 minAmountOut, address to)
+        external
+        payable
+        returns (uint256 amountOut, int256 deltaLiquidity);
+    function swapExactSyForPt(uint256 amountIn, uint256 minAmountOut, address to)
+        external
+        returns (uint256 amountOut, int256 deltaLiquidity);
+    function allocate(uint256 deltaX, uint256 deltaY, uint256 minLiquidityOut, address to)
+        external
+        returns (uint256 deltaLiquidity);
+    function prepareSwapSyIn(uint256 amountIn, uint256 timestamp, PYIndex index)
+        external
+        view
+        returns (uint256 amountInWad, uint256 amountOutWad, uint256 amountOut, int256 deltaLiquidity, uint256 strike_);
+    function prepareSwapPtIn(uint256 ptIn, uint256 timestamp, PYIndex index)
+        external
+        view
+        returns (uint256 amountInWad, uint256 amountOutWad, uint256 amountOut, int256 deltaLiquidity, uint256 strike_);
+}
+
+contract LiquidityManager {
+    using PYIndexLib for PYIndex;
+    using PYIndexLib for IPYieldToken;
+    using FixedPointMathLib for uint256;
+
+    function mintSY(address SY, address receiver, address tokenIn, uint256 amountTokenToDeposit, uint256 minSharesOut)
+        public
+        payable
+        returns (uint256 amountOut)
+    {
+        return _mintSYFromNativeAndToken(SY, receiver, tokenIn, amountTokenToDeposit, minSharesOut);
+    }
+
+    function _mintSYFromNativeAndToken(address SY, address receiver, address tokenIn, uint256 amountTokenIn, uint256 minSyMinted)
+        internal
+        returns (uint256 amountSyOut)
+    {
+        IStandardizedYield sy = IStandardizedYield(SY);
+        if (!sy.isValidTokenIn(tokenIn)) revert InvalidTokenIn(tokenIn);
+
+        if (msg.value > 0 && sy.isValidTokenIn(address(0))) {
+            // SY minted check is done in this function instead of relying on the SY contract's deposit().
+            amountSyOut += sy.deposit{value: msg.value}(address(this), address(0), msg.value, 0);
+        }
+
+        if (tokenIn != address(0)) {
+            ERC20(tokenIn).transferFrom(msg.sender, address(this), amountTokenIn);
+            amountSyOut += sy.deposit(receiver, tokenIn, amountTokenIn, 0);
+        }
+
+        if (amountSyOut < minSyMinted) {
+            revert InsufficientSYMinted(amountSyOut, minSyMinted);
+        }
+    }
+
+    function allocateFromSy(IRMM rmm, uint256 amountSy, uint256 minPtOut, uint256 minLiquidityDelta, uint256 initialGuess, uint256 epsilon) external returns (uint256 liquidity) {
```

Comment:
Address arguments should be addresses, not interfaces. interfaces make it clunky

Reply:
Multiple places with this